velox/functions/lib/aggregates/WeightedAverageAggregateBase.h

```diff
@@ -0,0 +1,524 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/exec/Aggregate.h"
+#include "velox/functions/lib/aggregates/DecimalAggregate.h"
+#include "velox/type/DecimalUtil.h"
+#include "velox/vector/ComplexVector.h"
+#include "velox/vector/DecodedVector.h"
+#include "velox/vector/FlatVector.h"
+
+namespace facebook::velox::functions::aggregate {
+
+namespace {
+
+/// Translate selected rows of decoded to the corresponding rows of its base
+/// vector.
+SelectivityVector translateToInnerRows(
```

Comment:
Why not use/update the one in `SelectivityVector.h`?

Reply:
I actually copied this function from `AverageAggregateBase.h` .
Should we refactor this into `SelectivityVector.h` ?